cli/proc_state.rs

```diff
@@ -712,17 +552,38 @@ impl ProcState {
       is_dynamic
     );
 
-    let graph_data = self.graph_data.lock();
-    let found_specifier = match graph_data.modules.get(&specifier) {
-      Some(ModuleEntry::Redirect(s)) => s,
-      _ => &specifier,
-    };
-    match graph_data.modules.get(found_specifier) {
-      Some(ModuleEntry::Module { code, .. }) => Ok(ModuleSource {
-        code: code.clone(),
-        module_url_specified: specifier.to_string(),
-        module_url_found: found_specifier.to_string(),
-      }),
+    let graph_data = self.graph_data.read();
+    let found = graph_data.follow_redirect(&specifier);
+    match graph_data.get(&found) {
+      Some(ModuleEntry::Module {
+        code, media_type, ..
+      }) => {
+        let code = match media_type {
+          MediaType::JavaScript
+          | MediaType::Unknown
+          | MediaType::Cjs
+          | MediaType::Mjs => code.as_ref().clone(),
+          MediaType::Dts => "".to_string(),
+          _ => {
+            let emit_path = self
+              .dir
+              .gen_cache
+              .get_cache_filename_with_extension(&found, "js")
+              .unwrap_or_else(|| {
+                panic!("Unable to get cache filename: {}", &found)
```

Comment:
Why `panic!` instead of returning an error?

Reply:
This happens when there's an unexpected protocol, maybe a cause for confusion that I didn't use `unreachable!()` instead.